pkg/plugin/datasource.go

```diff
@@ -0,0 +1,326 @@
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/criblcloud/search-datasource/pkg/models"
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// Ensure it actually implements the interfaces we need it to
+var (
+	_ backend.QueryDataHandler    = (*Datasource)(nil)
+	_ backend.CheckHealthHandler  = (*Datasource)(nil)
+	_ backend.CallResourceHandler = (*Datasource)(nil)
+)
+
+const MAX_RESULTS = 10000 // same as what the actual Cribl UI imposes
+const QUERY_PAGE_SIZE = 1000
+const CRIBL_TIME_FIELD = "_time"
+const MAX_DELAY_WAITING_FOR_FINISHED = 60 * time.Second // could make this configurable in a future release
```

Comment:
Is this for all queries, including those driving alerts? If so, wondering if eventually we may want two max delays - a second one for the background queries driving alerts, etc.

Reply:
All queries.  AFAIK there's no way to know whether the query is interactive or not.  I'll look into it to confirm.